src/protocol/frame/mod.rs

```diff
@@ -266,9 +269,9 @@ mod tests {
     fn size_limit_hit() {
         let raw = Cursor::new(vec![0x82, 0x07, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07]);
         let mut sock = FrameSocket::new(raw);
-        assert_eq!(
-            sock.read_frame(Some(5)).unwrap_err().to_string(),
-            "Space limit exceeded: Message length too big: 7 > 5"
-        );
+        match sock.read_frame(Some(5)) {
+            Err(Error::Capacity(CapacityError::MessageTooLong { size: 7, max_size: 5 })) => {}
+            _ => panic!(),
+        }
```

Comment:
nit: these assertions could be AFAIK expressed more succinctly by
```suggestion
        assert!(matches!(sock.read_frame(Some(5)), Err(Error::Capacity(CapacityError::MessageTooLong { size: 7, max_size: 5 }))));
```
I believe that `cargo clippy --all-targets` would suggest this, too?

Reply:
That's weird, clippy didn't pick up on that even with `--all-targets`. I wasn't aware of `matches!` - thanks for pointing that out.